### PR TITLE
chore: Bump Snaps packages

### LINKED ISSUE #2622: Add `snap_getPreferences`
Add `snap_getPreferences` as an alternative to `snap_getLocale` that returns both the user's preferred locale as well as their preferred currency.

### PR DESCRIPTION
**Pull request by Mrtenz**
## **Description**

This bumps all Snaps packages to the latest versions and handles breaking changes. It also introduces some new components and RPC methods.

Summary of changes in the snaps deps:
- Add new `snap_dialog` confirmation type that uses fully custom UI
- Add `Container` and `Footer` component
- Add `RadioGroup` component
- Add `Card` component
- Add `snap_getPreferences` to return user preferences
- Add `snap_resolveInterface` to resolve existing interfaces
- Handle initial connections for preinstalled Snaps
- Enforce stricter size limits for source code, static auxiliary files, and `snap_getFile` return values
- Account for revoked origins when calculating the difference in `initialConnections`

Closes https://github.com/MetaMask/MetaMask-planning/issues/1571
Closes https://github.com/MetaMask/MetaMask-planning/issues/2125
Closes https://github.com/MetaMask/snaps/issues/2622
Closes https://github.com/MetaMask/MetaMask-planning/issues/1956

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26086?quickstart=1)

### CODE REVIEW

File: .yarn/patches/@metamask-snaps-utils-npm-7.7.0-2cc1f044af.patch
Comment (GuillaumeRx) [inline .yarn/patches/@metamask-snaps-utils-npm-7.7.0-2cc1f044af.patch:1]: you need to bump `@metamask/snaps-sdk` under `resolutions` (l. 223)

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: .yarn/patches/@metamask-snaps-utils-npm-7.7.0-2cc1f044af.patch
Comment (GuillaumeRx) [reply .yarn/patches/@metamask-snaps-utils-npm-7.7.0-2cc1f044af.patch:1]: Or remove it ?

File: package.json
Comment (GuillaumeRx) [inline package.json:1]: You need to bump `@metamask/snaps-sdk` in the `resolutions` (l.223)

Comment (GuillaumeRx) [reply package.json:1]: Or remove it ?

**Review by GuillaumeRx — DISMISSED** [review]

File: package.json
Comment (Mrtenz) [reply package.json:1]: Done. We can't remove it because some packages pull in outdated versions.

**Comment by Mrtenz** [comment]
@SocketSecurity ignore npm/@metamask/snaps-utils@7.8.1

This is ours.

**Comment by FrederikBolding** [comment]
@SocketSecurity ignore npm/@metamask/snaps-utils@8.0.0

Network access expected.

**Comment by FrederikBolding** [comment]
https://github.com/SocketSecurity ignore npm/@metamask/snaps-utils@8.0.1

Network access expected.

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -3887,6 +3887,15 @@
...
   },
+  "permission_getPreferences": {
+    "message": "See information like your preferred language and fiat currency.",
+    "description": "The description for the `snap_getPreferences` permission"
+  },
+  "permission_getPreferencesDescription": {
+    "message": "Let $1 access information like your preferred language and fiat currency in your MetaMask settings. This helps $1 display content tailored to your preferences. ",
+    "description": "An extended description for the `snap_getPreferences` permission. $1 is the snap name."
+  },
+
```
Comment (Mrtenz) [inline app/_locales/en/messages.json:3898]: Seems to be an unnecessary whitespace here.

**Review by david0xd — DISMISSED** [review]

File: app/_locales/en/messages.json (hunk 1 above)
Comment (FrederikBolding) [reply app/_locales/en/messages.json:3898]: Good catch, fixed!

(curious how the linter doesn't complain about this)

**Review by Mrtenz — COMMENTED** [review]
Looks good to me. Can't approve my own PR.

**Review by david0xd — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]
Approving for @Mrtenz - given that I took over this PR

**Review by jpuri — APPROVED** [review]
Changes in files owned by confirmations team look good 👍
